src/applications/personalization/profile360/tests/service-history.pact.spec.js

```diff
@@ -0,0 +1,117 @@
+    it('responds with an array of serviceHistory objects when there is a service history', async () => {
+      await mockApi().addInteraction({
+        state: 'at least one entry in the service history exists',
+        uponReceiving: 'a GET request',
+        withRequest: {
+          method: 'GET',
+          path: '/v0/service_history',
+          headers: {
+            'X-Key-Inflection': 'camel',
+          },
+        },
+        willRespondWith: {
+          status: 200,
+          headers: {
+            'Content-Type': term({
+              matcher: '^application/json',
+              generate: 'application/json',
+            }),
+          },
+          body: {
+            data: {
+              attributes: {
+                serviceHistory: eachLike({
+                  branchOfService: string('Air Force'),
+                  beginDate: string('1900-01-31'),
+                  endDate: string('2000-12-25'),
+                }),
+              },
+            },
+          },
+        },
+      });
+
+      const dispatch = sinon.stub();
+      await fetchMilitaryInformation()(dispatch);
+    });
+    // user 1 will have this behavior
+    it('responds with an empty serviceHistory array when there is no service history', async () => {
+      await mockApi().addInteraction({
+        state: 'there are no service history records',
+        uponReceiving: 'a GET request',
+        withRequest: {
+          method: 'GET',
+          path: '/v0/service_history',
+          headers: {
+            'X-Key-Inflection': 'camel',
+          },
+        },
+        willRespondWith: {
+          status: 200,
+          headers: {
+            'Content-Type': term({
+              matcher: '^application/json',
+              generate: 'application/json',
+            }),
+          },
+          body: {
+            data: {
+              attributes: {
+                serviceHistory: [],
+              },
+            },
+          },
+        },
+      });
+
+      const dispatch = sinon.stub();
+      await fetchMilitaryInformation()(dispatch);
+    });
+    // user 10 will have this behavior
+    it('responds with a 403 error when the user is not a Veteran', async () => {
+      await mockApi().addInteraction({
+        state: 'not a Veteran',
+        uponReceiving: 'a GET request',
+        withRequest: {
+          method: 'GET',
+          path: '/v0/service_history',
+          headers: {
+            'X-Key-Inflection': 'camel',
+          },
+        },
+        willRespondWith: {
+          status: 403,
+          headers: {
+            'Content-Type': term({
+              matcher: '^application/json',
+              generate: 'application/json',
+            }),
+          },
+          // What we really want to test is: "the `errors` array contains at
+          // least one object that has a code value of '403'". I think this test
+          // will fail if `errors` contains more than one element
```

Comment:
@U-DON I'd appreciate any thoughts you have re: this comment!

Reply:
Would the `eachLike` [matcher](https://github.com/pact-foundation/pact-js#match-based-on-arrays) help here?

Reply:
Ah, okay. I don't think the current spec of Pact has matchers that are flexible enough for that.

But [this feature request](https://github.com/pact-foundation/pact-specification/issues/38) for something like `arrayContains` seems like it would've been useful here.

So it looks like the best we can do here is just use `eachLike` without the guarantee that at least one element has a code 403.